src/containers/shared/components/Transaction/SetHook/parser.ts

```diff
@@ -0,0 +1,23 @@
+import { SetHook, SetHookInstructions } from './types'
+
+export const parser = (tx: SetHook, meta: any): SetHookInstructions => {
+  const hooks = tx.Hooks.map((hook) => hook.Hook)
+  const affectedNodes = meta.AffectedNodes.filter(
+    (node: any) =>
+      node.CreatedNode?.LedgerEntryType === 'Hook' ||
+      (node.ModifiedNode?.LedgerEntryType === 'Hook' &&
+        !!node.ModifiedNode?.PreviousFields?.Hooks),
+  )
+  const hashes = affectedNodes.flatMap((node: any) =>
+    (node.ModifiedNode?.FinalFields ?? node.CreatedNode?.NewFields)?.Hooks?.map(
+      (hook: any) => hook.Hook.HookHash,
+    ),
+  )
+  // TODO: there may be bugs here when a `HookHash` is already specified in a hook
+  // It's difficult to understand what situation that would be in, so this is left here for now
```

Comment:
is there a test for this?

Reply:
No, because I don't know what situations this can arise in (the docs aren't specific enough)
That's why I left a TODO